Realm/RLMQueryUtil.mm

```diff
@@ -1850,6 +1852,34 @@ bool is_self_value_for_key_path_function_expression(NSExpression *expression)
                        @"Only support compound, comparison, and constant predicates");
     }
 }
+
+// For nested subscripts in NSPredicate like `anyCol[0]['key'] we need to iterate through the nested NSExpression to get the paths of the query.
+void QueryBuilder::get_path_elements(std::vector<PathElement> &paths, NSExpression *expression) {
+    for (int i = 0; i < expression.arguments.count; i++) {
+        if ((expression.arguments.count > 1) && (expression.arguments[i].expressionType == NSFunctionExpressionType)) {
+            get_path_elements(paths, expression.arguments[0]);
+        } else {
+            if (expression.arguments[i].expressionType == NSConstantValueExpressionType) {
+                id value = [expression.arguments[i] constantValue];
+                if ([value isKindOfClass:[NSNumber class]]) {
+                    NSNumber *index = (NSNumber *)value;
+                    if (index) {
+                        paths.push_back(PathElement{[index intValue]});
+                    }
+                } else if ([value isKindOfClass:[NSString class]]) {
+                    NSString *key = (NSString *)value;
+                    if ([key isEqual:@"all"]) {
```

Comment:
???

If you have a field named "all" it just completely ignores the field name and does a wildcard search instead?

Reply:
Why is this using "all" as the wildcard search key?